packages/plugin-ext/src/main/browser/view/tree-views-main.tsx

```diff
@@ -218,9 +226,14 @@ export class TreeViewWidget extends TreeWidget {
         @inject(TreeProps) readonly treeProps: TreeProps,
         @inject(TreeModel) readonly model: TreeModel,
         @inject(ContextMenuRenderer) readonly contextMenuRenderer: ContextMenuRenderer,
-        @inject(TreeViewDataProviderMain) readonly dataProvider: TreeViewDataProviderMain) {
+        @inject(TreeViewDataProviderMain) readonly dataProvider: TreeViewDataProviderMain,
+        @inject(SelectionService) readonly selectionService: SelectionService) {
 
         super(treeProps, model, contextMenuRenderer);
+
+        this.model.onSelectionChanged(selection => {
+            this.selectionService.selection = selection;
```

Comment:
only if a tree view has focus, like here: https://github.com/theia-ide/theia/blob/bb6700b60dfc1951df98844bdbb2e072f1e01716/packages/navigator/src/browser/navigator-widget.tsx#L74-L76

Reply:
Temporary added a todo comment, need to resolve this issue to have an ability to check active widget in complex views: https://github.com/theia-ide/theia/issues/4623